data_structures/radix_sort/radix_sort.py

```diff
@@ -0,0 +1,43 @@
+def radix_sort(arr):
+    """
+    Takes a list of positive unordered integers and returns an
+    ascending sorted list using radix_sort.
+    """
+    if type(arr) is not list:
+        return TypeError('Must use a list of positive integers.')
+
+    for item in arr:
+        if item < 0:
+            return TypeError('Must not contain negative integers.')   
+
+    iteration = 0  # used to select digit place
+   
+    while iteration <= len(str(max(arr))):
```

Comment:
Since this loops for a set number of iterations, you should be using a `for` loop instead of a `while` loop.

Reply:
Also, this condition gets evaluated each time the loop occurs, thus you are calculating the `max` of the list each iteration, which is rather inefficient.